src/scene/materials/custom-material.js

```diff
@@ -0,0 +1,72 @@
+import { ShaderProcessorOptions } from '../../platform/graphics/shader-processor-options.js';
+import { LIGHTTYPE_DIRECTIONAL, LIGHTTYPE_OMNI, LIGHTTYPE_SPOT, MASK_AFFECT_DYNAMIC, SHADERDEF_INSTANCING, SHADERDEF_MORPH_NORMAL, SHADERDEF_MORPH_POSITION, SHADERDEF_MORPH_TEXTURE_BASED, SHADERDEF_SCREENSPACE, SHADERDEF_SKIN } from "../constants";
+import { getProgramLibrary } from "../shader-lib/get-program-library";
+import { LitOptions } from "./lit-options";
+import { collectLights } from "./lit-material-common.js";
+import { Material } from './material.js';
+import { custom } from '../shader-lib/programs/custom.js';
+
+/**
+ * A custom material provides a custom front-end part of a lit material
+ *
+ * @augments Material
+ */
+
+class CustomMaterial extends Material {
+    _litOptions = new LitOptions();
+
+    _argumentsChunk = "";
+
+    set litOptions(value) {
+        this._litOptions = value;
+    }
+
+    get litOptions() {
+        return this._litOptions;
+    }
+
+    set argumentsChunk(value) {
```

Comment:
this should have JSDocs as it's a public API

Reply:
I'd be keen to find some better name here. Perhaps `customChunk`, to match the CustomMaterial. @willeastcott any suggestions? Could also be `materialChunk`, I think I like this even better, or maybe even just `chunk`.